app/forms/vpc-router-route-common.tsx

```diff
@@ -74,3 +88,34 @@ export const targetValueDescription = (targetType: RouteTarget['type']) =>
   targetType === 'internet_gateway'
     ? routeFormMessage.internetGatewayTargetValue
     : undefined
+
+type RouteFormFieldsProps = {
+  control: Control<RouteFormValues>
+  targetType: RouteTarget['type']
+  isDisabled?: boolean
+}
+export const RouteFormFields = ({
+  control,
+  targetType,
+  isDisabled,
+}: RouteFormFieldsProps) => (
+  <>
+    {isDisabled && (
+      <Message variant="info" content={routeFormMessage.vpcSubnetNotModifiable} />
+    )}
+    <NameField name="name" control={control} disabled={isDisabled} />
+    <DescriptionField name="description" control={control} disabled={isDisabled} />
+    <ListboxField {...fields.destType} control={control} disabled={isDisabled} />
```

Comment:
If `fields` is no longer shared, you can get rid of it and inline these as regular props.